python-lib/dku_config/dku_model_handler.py

```diff
@@ -53,21 +53,52 @@ def _compute_features(self):
         self.candidate_features = [feature for feature in non_excluded_features if self.features[feature]['role'] == 'REJECT']
 
     def _compute_base_values(self):
-        """Calculates base values for each used feature, considering their type."""
+        """Calculates base values for each used feature, considering their type and computes weighted average for categorical variables if applicable."""
         self.base_values = {}
         collector_data = self.model_info_handler.get_collector_data()['per_feature']
+        exposure_column = self.exposure_columns[0] if self.exposure_columns else None
+        train_df = self.model_info_handler.get_train_df() if exposure_column and exposure_column in self.model_info_handler.get_train_df().columns else None
+
         for feature in self.used_features:
             feature_data = collector_data[feature]
-            if self.features[feature]['type'] == 'CATEGORY':
-                # Use .get to avoid KeyError if 'dropped_modality' key is missing
-                self.base_values[feature] = feature_data.get('dropped_modality')
-                if self.base_values[feature] is None:
-                    # Handle the case where 'dropped_modality' is not available
-                    raise ValueError(f"Warning: 'dropped_modality' not found for feature {feature}. Please ensure drop one dummy is enabled and clipping uses max nb categories")
-            elif self.features[feature]['type'] == 'NUMERIC':
-                self.base_values[feature] = feature_data['stats']['average']
+            feature_type = self.features[feature]['type']
+
+            if feature_type == 'CATEGORY':
+                if exposure_column and train_df is not None:
```

Comment:
I don't understand what we are doing here. Are we not always taking the dropped modality as the base value?

Reply:
We should do yes, lines 70-73 updated to reflect this